### PR TITLE
Improves shield generator code

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
bugfix: Fixed a bug where getting caught between two shield generators as they activated wouldn't cause you to gib and die.
/:cl:

### CODE REVIEW

**Comment by Supermichael777** [comment]
10/10 lost as ai due to fail

**Comment by WJohn** [comment]
Huh, I wonder if people would make a killing hallway sort of thing with these now, by having loads of shield gens and then having a rogue AI or rogue cyborg activate them at range, trapping them in and then gibbing them.

**Comment by Kromgar** [comment]
@WJohn setup on that would be a bitch also you cant turn it off without cutting power

File: code/__HELPERS/game.dm [hunk 1]
```diff
@@ -535,4 +535,14 @@
 	return hex2num(copytext(hexa, 4, 6))
 
 /proc/GetBluePart(const/hexa)
-	return hex2num(copytext(hexa, 6, 8))
\ No newline at end of file
+	return hex2num(copytext(hexa, 6, 8))
+
+/proc/get_opposite_dir(direction) //returns the direction opposite the input direction.
```
Comment (RemieRichards) [inline code/__HELPERS/game.dm]: why do people always define a proc for this?

```turn(dir, 180)``` works in all cases (I don't know about UP, DOWN, but I suspect it probably does, not to mention we don't use those (yet))

Comment (ChangelingRain) [inline code/__HELPERS/game.dm]: everyone forgets about turn, rip me

File: code/game/machinery/shieldgen.dm [hunk 2]
```diff
@@ -489,37 +439,19 @@
...
-
+	if(needs_power && gen_primary)
+		gen_primary.use_stored_power(drain_amount*0.5)
+		if(gen_secondary) //using power may cause us to be destroyed
+			gen_secondary.use_stored_power(drain_amount*0.5)
 
 /obj/machinery/shieldwall/CanPass(atom/movable/mover, turf/target, height=0)
-	if(height==0) return 1
+	if(height==0)
+		return 1
```
Comment (lzimann) [inline code/game/machinery/shieldgen.dm]: `TRUE`

**Pull request merged**